algopy_testing/tests/primitives/test_biguint.py

```diff
@@ -0,0 +1,449 @@
+import operator
+import re
+from collections.abc import Callable
+
+import algokit_utils
+import pytest
+from algopy import BigUInt
+from algopy.constants import MAX_UINT64, MAX_UINT512
+
+from tests.primitives.conftest import AVMInvoker
+
+_negative_value_error = "expected positive value"
+_too_big512_error = re.escape(f"expected value <= {MAX_UINT512}")
+
+_underflow_error = "- underflows"
+
+_avm_underflow_error = "byte math would have negative result"
+_avm_zero_division_error = "division by zero"
+_avm_zero_mod_error = "modulo by zero"
+
+
+def _avm_overflow_error() -> str:
+    return re.escape("assert failed")
+
+
+@pytest.mark.parametrize(
+    "value",
+    [
+        0,
+        1,
+        MAX_UINT512,
+    ],
+)
+def test_biguint_unary_plus(value: int) -> None:
+    # just test the implementation as there is no AVM equivalent
+    assert +BigUInt(value) == value
+
+
+@pytest.mark.parametrize(
+    "value",
+    [
+        0,
+        1,
+        MAX_UINT512,
+    ],
+)
+def test_biguint_bool(value: int) -> None:
+    # just test the implementation as there is no AVM equivalent
+    assert bool(BigUInt(value)) == bool(value)
+
+
+@pytest.mark.parametrize(
+    ("a", "b"),
+    [
+        (0, 0),
+        (0, 1),
+        (0, MAX_UINT512),
+        (1, 0),
+        (1, 1),
+        (1, MAX_UINT512),
+        (MAX_UINT512, MAX_UINT512),
+    ],
+)
+@pytest.mark.parametrize(
+    "op_name",
+    [
+        "eq",
+        "ne",
+        "lt",
+        "le",
+        "gt",
+        "ge",
+    ],
+)
+def test_biguint_comparison(get_avm_result: AVMInvoker, op_name: str, a: int, b: int) -> None:
+    avm_result = get_avm_result(f"verify_biguint_{op_name}", a=a, b=b)
+    op = getattr(operator, op_name)
+
+    assert avm_result == op(BigUInt(a), BigUInt(b))
```

Comment:
Most of these operations should also have some tests that compare mixed operands types with `BigUint` and `UInt64` 

e.g `BigUint(a) < UInt64(b)` and `UInt64(a) < BigUint(b)`

Reply:
I've added those tests.